tests/Toolkit/ATest.php

```diff
@@ -698,4 +698,54 @@ public function testWrap()
         $result = A::wrap(null);
         $this->assertSame([], $result);
     }
+
+
+     /**
+     * @covers ::where
+     */
+    public function testWhere()
+    {
+        $associativeArray = $this->_array();
+        $array = array_keys($associativeArray);
+
+        $result = A::filter($associativeArray, function($value, $key){
+            return in_array($key, ['cat', 'dog']);
+        });
+        $this->assertSame(['cat'  => 'miao', 'dog'  => 'wuff'], $result);
+
+        $result = A::filter($associativeArray, function($value, $key){
+            return in_array($value, ['miao', 'tweet']);
+        });
+        $this->assertSame(['cat'  => 'miao', 'bird' => 'tweet'], $result);
+
+        $result = A::filter($associativeArray, function($value, $key){
+            return $key === 'cat' || $value === 'tweet';
+        });
+        $this->assertSame(['cat'  => 'miao', 'bird' => 'tweet'], $result);
+
+        $result = A::filter($array, function($value, $key){
+            return $key > 0;
+        });
+        $this->assertSame([1 => 'dog', 2 => 'bird'], $result);
+    }
+
+     /**
+     * @covers ::without
+     */
+    public function testWithout()
+    {
+        $associativeArray = $this->_array();
+        $array = array_keys($associativeArray);
+
+        $array = [...array_keys($associativeArray), ...array_keys($associativeArray)];
+
+        $this->assertSame(['dog' => 'wuff', 'bird' => 'tweet'], A::without($associativeArray, 'cat'));
+        $this->assertSame(['dog' => 'wuff'], A::without($associativeArray, ['cat', 'bird']));
+        $this->assertSame([], A::without($associativeArray, ['cat', 'dog', 'bird']));
+        $this->assertSame($associativeArray, A::without($associativeArray, ['this', 'doesnt', 'exist']));
```

Comment:
Another case with a mixture of existing and non-existing methods:

```suggestion
        $this->assertSame(['dog' => 'wuff', 'bird' => 'tweet'], A::without($associativeArray, ['this', 'cat', 'doesnt', 'exist']));
```

Reply:
@lukasbestle Don't quite understand? Updated the tests a bit anyway (according to Github suggestion), but now I realise that might not have been what you meant.

Reply:
What I meant is that the test should also cover the case where the `$keys` include a key that exists but also keys that don't exist. This way the tests cover a broader range of cases to avoid future regressions.

I've committed my suggestion to the PR branch.

Reply:
@lukasbestle oh, you meant _keys_, right-o! Thanks for committing it